From 4cf34517b0c793405dca154cc70c9b5d1a566a8a Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Fri, 29 Sep 2023 14:52:50 -0400
Subject: [PATCH] vbox_fb: fix kernel v6.5 build

When building against the 6.5 kernel (and kernel headers), the following
build error is encountered:

   | vboxguestdrivers/7.0.10/vboxguestdrivers-7.0.10/vboxvideo/vbox_fb.c:194:24: error: 'drm_fb_helper_sys_fillrect' undeclared here (not in a function); did you mean 'drm_fb_helper_lastclose'?
   |   194 |         .fb_fillrect = drm_fb_helper_sys_fillrect,
   |       |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
   |       |                        drm_fb_helper_lastclose
   | vboxguestdrivers/7.0.10/vboxguestdrivers-7.0.10/vboxvideo/vbox_fb.c:195:24: error: 'drm_fb_helper_sys_copyarea' undeclared here (not in a function); did you mean 'drm_fb_helper_set_par'?
   |   195 |         .fb_copyarea = drm_fb_helper_sys_copyarea,
   |       |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
   |       |                        drm_fb_helper_set_par
   | vboxguestdrivers/7.0.10/vboxguestdrivers-7.0.10/vboxvideo/vbox_fb.c:196:25: error: 'drm_fb_helper_sys_imageblit' undeclared here (not in a function)
   |   196 |         .fb_imageblit = drm_fb_helper_sys_imageblit,
   |       |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~

We backport a patch from vbox-dev to fix the issue.

Upstream-Status: Submitted [https://www.mail-archive.com/vbox-dev@virtualbox.org/msg10129.html]

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 src/VBox/Additions/linux/drm/vbox_fb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/VBox/Additions/linux/drm/vbox_fb.c b/src/VBox/Additions/linux/drm/vbox_fb.c
index 2a8c79ab..d453533c 100644
--- a/src/VBox/Additions/linux/drm/vbox_fb.c
+++ b/src/VBox/Additions/linux/drm/vbox_fb.c
@@ -196,9 +196,13 @@ static struct fb_ops vboxfb_ops = {
 	.owner = THIS_MODULE,
 	.fb_check_var = drm_fb_helper_check_var,
 	.fb_set_par = drm_fb_helper_set_par,
+#if RTLNX_VER_MIN(6, 5, 0)
+	FB_DEFAULT_SYS_OPS,
+#else
 	.fb_fillrect = drm_fb_helper_sys_fillrect,
 	.fb_copyarea = drm_fb_helper_sys_copyarea,
 	.fb_imageblit = drm_fb_helper_sys_imageblit,
+#endif
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
-- 
2.34.1

